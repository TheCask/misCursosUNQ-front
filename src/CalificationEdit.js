import React, { useState, useEffect } from 'react';
import { Input, Table, Form, ButtonGroup, InputGroup, InputGroupAddon, Button, Row, Col} from 'reactstrap';
import CRUDSaveButton from './CRUDButtonBar/CRUDSaveButton';
import * as CourseAPI from './services/CourseAPI';
import Log from './Log';


export default function CalificationEdit(props){
    
    const [dirty, setDirty] = useState(false)
    const [califificationFullList, setCalifFullList] = useState([]) 
    const [filterValue, setFilterValue] = useState('');

    Log.info(califificationFullList, "Calification Picker")

    useEffect(() => {
        if (props.currEvalInstance){
            setCalifFullList(createCalifFullList(props.students, props.currEvalInstance.califications))
        }
    }, [props.currEvalInstance])

    function handleNoteChange(event){
        setDirty(true);
        const newNote = event.target.value;
        setCalifFullList(califificationFullList.map( cal => 
            cal.student.fileNumber === parseInt(event.target.name) ? {...cal, note: newNote} :  cal))
    }

    function createCalifFullList(students, califications){
        // Uses student list and existing califications to create califications for all students
        if (props.currEvalInstance){
            const sl = students.map( st => {
                const calificationFound = califications.find( cal => cal.student.fileNumber === st.fileNumber);
                return ( populateCalif(st, calificationFound));
            })
            return sl;
        }    
        else {
            return [];
        }
    }

    function populateCalif (st, cal){
        if (cal){
            cal.student = st
            return cal;
        } else{
            return {
                calificationId: '',
                student: { ...st},
                note: ''
            }
        }
    }

    function onSaveCalifications(event){
        event.preventDefault()
        Log.info("Saving califications...........")
        const evalWithCalifications = {...props.currEvalInstance, califications: califificationFullList}
        CourseAPI.postCourseEvaluationAsync(
            props.courseId, 
            evalWithCalifications, 
            (res) => {
                Log.info(res, "POST SUCCEDED");
                setDirty(false);
                props.reloadCourse();

            }, 
            props.showError("save course")
        );

    }

    function isEvalSelected() { 
        return props.currEvalInstance && props.currEvalInstance.evaluationId !== '';
    }

    function getInnerPropValue(baseObj, subPropString){
        const subProps = subPropString.split(".");
        const lastPropName = subProps.pop(); // elimina del array y retorna el ultimo 
        let propRef = baseObj;
        subProps.forEach(subprop => {
          propRef = propRef[subprop];
        });
        return propRef[lastPropName];
    }

    const tableRender = (itemList) => {
        const filteredFields = ["student.personalData.lastName", "student.personalData.firstName", "student.fileNumber", "note"];
        let filteredList = itemList;
        if (filterValue !== '')
            filteredList = itemList.filter( cal => 
                filteredFields.some( k => {
                    Log.info(getInnerPropValue(cal, k), "getInnerPropValue(cal, k)");
                    return getInnerPropValue(cal, k) ? getInnerPropValue(cal, k).toString().toLowerCase().includes(filterValue.toLowerCase()) : false
                
                })             
            )
        return <Table hover className="mt-4" scrollable={"true"} >
            <thead>
                <tr>
                    <th width="25%">File Number</th>
                    <th width="50%">Student Name</th>
                    <th width="25%">Grade</th>
                </tr>
            </thead>
            <tbody>
                {filteredList.map( (item, index) => 
                    <tr key={index} >
                        <td style={{whiteSpace: 'nowrap'}}>{item.student.fileNumber || ''}</td>
                        <td style={{whiteSpace: 'nowrap'}}>{`${item.student.personalData.lastName}, ${item.student.personalData.firstName}` || ''}</td>
                        <td style={{whiteSpace: 'nowrap'}}>
                            <Input 
                                id={item.evaluationId}
                                name = {item.student.fileNumber}
                                value={item.note || ''} 
                                onChange={handleNoteChange}
                                disabled={!props.currEvalInstance}
                                type="float"  min="0.0" max="10.0"
                                title="A number between 0 and 10, with up to 2 decimals"
                                pattern={"^(10|10.0|10.00|([0-9])([.][0-9]([0-9])?)?)$"}
                                />
                        </td>
                    </tr>
                )}
            </tbody>
        </Table>;
    }
    
    
    return (
        <>
        <Form onSubmit={onSaveCalifications}>
            <Row>
                <Col md={10}>
                <InputGroup>
                    <Input 
                        placeholder={"Search student..."} 
                        value={filterValue} 
                        onChange={(event) => setFilterValue(event.target.value)} 
                        disabled = {!isEvalSelected()}
                        />
                    <InputGroupAddon addonType="append">
                        <Button color="secondary" onClick={() => setFilterValue('')} >Clear</Button>
                    </InputGroupAddon>
                </InputGroup>
                </Col>
                <Col>
                    <ButtonGroup style={{ padding: "0em", float: "right"}}>
                    <CRUDSaveButton
                        behavior = {{type: "submit"}}
                        entityTypeCapName = {'Calification'}
                        isDisabled = {!isEvalSelected() || !dirty}
                        />
                    </ButtonGroup>
                </Col>
            </Row>
            {tableRender(califificationFullList || [])}
        </Form>
        </>
    )
}
