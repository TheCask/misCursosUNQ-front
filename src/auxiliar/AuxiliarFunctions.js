
export function setInnerPropValue(baseObj, subPropString, value) {
    const subProps = subPropString.split(".");
    const lastPropName = subProps.pop(); // elimina del array y retorna el ultimo 
    let propRef = baseObj
    subProps.forEach(subprop => {
      propRef = propRef[subprop];
    });
    propRef[lastPropName] = value;
}