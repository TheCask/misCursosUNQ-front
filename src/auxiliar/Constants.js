export const userCsvToJsonMap = {
    dni: 'personalData.dni',
    firstName: 'personalData.firstName',
    lastName: 'personalData.lastName',
    email: 'personalData.email',
    cellPhone: 'personalData.cellPhone',
    cuitNumber: 'jobDetail.cuitNumber',
    category: 'jobDetail.category',
    grade: 'jobDetail.grade',
    dedication: 'jobDetail.dedication',
    contractRelation: 'jobDetail.contractRelation',
    aditionalHours: 'jobDetail.aditionalHours',
    cvURL: 'jobDetail.cvURL',
    lastUpdate: 'jobDetail.lastUpdate',
    gradeTitles: 'jobDetail.gradeTitles',
    posGradeTitles: 'jobDetail.posGradeTitles'
};

export const studentsCsvToJsonMap = {
    fileNumber: 'fileNumber',
    dni: 'personalData.dni',
    firstName: 'personalData.firstName', 
    lastName: 'personalData.lastName', 
    email: 'personalData.email',
    cellPhone: 'personalData.cellPhone'
};

export const CategoryOptions = ['', 'Auxiliar', 'Intructor/a', 'Adjunto/a', 'Asociado/a', 'Titular', 'Emérito/a', 'Consulto/a']
export const GradeOptions = ['', 'A', 'B']
export const ContractOptions = ['', 'Contratado/a', 'Interino/a', 'Ordinario/a']
export const DedicationOptions = ['', 'Parcial', 'Semi-Exclusiva', 'Exclusiva']

export const SeasonOptions = ['1C', '2C', '3C', '1T', '2T', '3T', '4T', '1S', '2S']
export const ShiftOptions = ['Mañana', 'Tarde', 'Noche']

export const emptyUser = {
    isActive: true,
    personalData: {
        dni: '',
        firstName: '',
        lastName: '',
        email: '',
        cellPhone: '' 
    },
    jobDetail: {
        cuitNumber: '',
        category: '',
        grade: '',
        dedication: '',
        contractRelation: '',
        aditionalHours: 0,
        cvURL: '',
        lastUpdate: '',
        gradeTitles: '',
        posGradeTitles: ''
    },
    coordinatedSubjects: [],
    taughtCourses: []
};

export const emptyCourse = {
    courseCode: '',
    courseShift: '',
    courseIsOpen: '',
    courseYear: 0,
    courseSeason: '',
    courseLocation: '',
    subject: {
        code: ''
    },
    students: [],
    teachers: []
};

export const emptyNewCourse = {
    courseCode: '',
    courseShift: '',
    courseIsOpen: true,
    courseYear: 2020,
    courseSeason: '',
    courseLocation: '',
    subject: {
        code: '',
        name: ''
    },
    students: [],
    lessons: [],
    teachers: [],
    evaluations: []
};

export const profileUser = {
    birthDate: "",
    data: {}, // not used
    email: "",
    firstName: "",
    fullName: "", // Autoconstruct lastName, firstName
    imageUrl: "",
    lastName: "",
    middleName: "", // Not Used
    mobilePhone: "",
    passwordChangeRequired: false,
    preferredLanguages: [], // this property always concats to the list (PATCH method bug on FusionAuth)
    timezone: "Etc/GMT-3", // not change
    twoFactorDelivery: "None", // not used
    twoFactorEnabled: false, // not used
    usernameStatus: "", // not used
    username: ""
};

export const emptyStudent = {
    fileNumber: 0,
    personalData: {
      // personalDataId: 0,
      dni: 0,
      firstName: '',
      lastName: '',
      email: '',
      cellPhone: ''
    },
    takenCourses: [],
    attendedLessons: []
};

export const emptySubject = {
    code: '',
    name: '',
    acronym: '',
    programURL: ''
};

export const parserConfig = {
    delimiter: "",  // auto-detect
    newline: "",  // auto-detect
    quoteChar: '"',
    escapeChar: '"',
    header: true,
    transformHeader: undefined,
    dynamicTyping: true,
    preview: 0,
    encoding: "",
    worker: true,
    comments: false,
    step: undefined,
    complete: undefined,
    error: undefined,
    download: false,
    downloadRequestHeaders: undefined,
    skipEmptyLines: false,
    chunk: undefined,
    fastMode: undefined,
    beforeFirstChunk: undefined,
    withCredentials: undefined,
    transform: undefined,
    delimitersToGuess: [',', '	', '|', ';']
};

export function newEvalInstance() {
    return {
        evaluationId: '',
        instanceName: 'TP ',
        califications: []
    }
}

export const initialStudent = () => { return { personalData: {} } }

export const initialUser = () => { return { isActive: true, personalData: {}, jobDetail: {} } }