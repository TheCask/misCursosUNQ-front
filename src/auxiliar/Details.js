import React, { Component } from 'react';
import UserEdit from '../user/UserEdit'
import StudentEdit from '../student/StudentEdit'
import SubjectEdit from '../subject/SubjectEdit'
import CourseEdit from '../course/CourseEdit'
import { userContext } from '../login/UserContext';
import AccessError from '../errorHandling/AccessError';

export class UserDetail extends Component {
  render() {
    this.actualRol = this.context.actualRol;
    return (this.actualRol === 'Guest' ?
      <AccessError errorCode="Guests are not allowed" 
          errorDetail="Make sure you are signed in with valid role before try to access this page"/>
      : 
      <UserEdit onlyDetail={true}/>
    )
  }
}

export class StudentDetail extends Component {
  render() {
    this.actualRol = this.context.actualRol;
    return (this.actualRol === 'Guest' ?
      <AccessError errorCode="Guests are not allowed" 
          errorDetail="Make sure you are signed in with valid role before try to access this page"/>
      : 
      <StudentEdit onlyDetail={true}/>
    )
  }
}

export class SubjectDetail extends Component {
  render() {
    this.actualRol = this.context.actualRol;
    return (this.actualRol === 'Guest' ?
      <AccessError errorCode="Guests are not allowed" 
          errorDetail="Make sure you are signed in with valid role before try to access this page"/>
      : 
      <SubjectEdit onlyDetail={true}/>
    )
  }
}

export class CourseDetail extends Component {
  render() {
    this.actualRol = this.context.actualRol;
    return (this.actualRol === 'Guest' ?
      <AccessError errorCode="Guests are not allowed" 
          errorDetail="Make sure you are signed in with valid role before try to access this page"/>
      : 
      <CourseEdit onlyDetail={true}/>
    )
  }
}

UserDetail.contextType = userContext;
StudentDetail.contextType = userContext;
SubjectDetail.contextType = userContext;
CourseDetail.contextType = userContext;