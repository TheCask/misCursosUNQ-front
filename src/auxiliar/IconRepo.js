import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default function getIcon(operationType, entityTypeCapName) {
    
    switch(operationType) {

        case 'Add' : 

            switch (entityTypeCapName) {
            
                case 'User': return ( <span>
                    <FontAwesomeIcon icon="chalkboard-teacher" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                case 'Teacher': return ( <span>
                    <FontAwesomeIcon icon="chalkboard-teacher" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                case 'Coordinator': return ( <span>
                    <FontAwesomeIcon icon="chalkboard-teacher" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                case 'Student': return ( <span>
                    <FontAwesomeIcon icon="user-graduate" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                case 'Subject': return ( <span>
                    <FontAwesomeIcon icon="university" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                case 'Course': return ( <span>
                    <FontAwesomeIcon icon="chalkboard" size="1x"/>
                    <FontAwesomeIcon icon="plus-circle" size="1x" transform="right-5 up-5"/>
                </span> )
                default: return (<FontAwesomeIcon icon='plus-circle' size="2x" style={{ width: "1em"}}/>)
            }

        case 'Edit' :

            switch (entityTypeCapName) {

                default : return <FontAwesomeIcon icon='edit' size="2x" style={{ width: "1em"}}/>
            }
            
        case 'Delete' :

            switch (entityTypeCapName) {

                default : return <FontAwesomeIcon icon='trash-alt' size="2x"/>
            }

        default : return (<FontAwesomeIcon icon='plus-circle' size="2x" style={{ width: "1em"}}/>)

    }
}

export function getCourseIcon(subjectCode) {
    switch(subjectCode.split("-")[0]) {
    case "80000":
        return (<> <FontAwesomeIcon icon='book' size="1x" color="darkred" transform="left-10 up-10"/>
                <FontAwesomeIcon icon='book-reader' size="1x" color="darkred" transform="right-10 up-10"/>
                <FontAwesomeIcon icon='pencil-alt' size="1x" color="darkred" transform="left-10 down-10"/>
                <FontAwesomeIcon icon='graduation-cap' size="1x" color="darkred" transform="right-10 down-10"/> </>);
    case "80005":
        return (<> <FontAwesomeIcon icon='bug' size="1x" color="black" transform="left-10 up-10"/>
                <FontAwesomeIcon icon='microchip' size="1x" color="black" transform="right-10 up-10"/>
                <FontAwesomeIcon icon='laptop-code' size="1x" color="black" transform="left-10 down-10"/>
                <FontAwesomeIcon icon='project-diagram' size="1x" color="black" transform="right-10 down-10"/> </>);
    case "80003":
        return (<> <FontAwesomeIcon icon='brain' size="1x" color="darkblue" transform="left-10 up-10"/>
                <FontAwesomeIcon icon='shapes' size="1x" color="darkblue" transform="right-10 up-10"/>
                <FontAwesomeIcon icon='infinity' size="1x" color="darkblue" transform="left-10 down-10"/>
                <FontAwesomeIcon icon='calculator' size="1x" color="darkblue" transform="right-10 down-10"/> </>);
    case "80004":
        return (<> <FontAwesomeIcon icon='thermometer-half' size="1x" color="darkgreen" transform="left-10 up-10"/>
                <FontAwesomeIcon icon='atom' size="1x" color="darkgreen" transform="right-10 up-10"/>
                <FontAwesomeIcon icon='flask' size="1x" color="darkgreen" transform="left-10 down-10"/>
                <FontAwesomeIcon icon='magnet' size="1x" color="darkgreen" transform="right-10 down-10"/> </>);
    default: return <FontAwesomeIcon icon={['fas', 'chalkboard']} size="2x" color="gray"/>
    }
}