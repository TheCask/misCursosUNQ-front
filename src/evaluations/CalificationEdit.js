import React, { useState, useEffect } from 'react';
import { withRouter } from 'react-router-dom';
import { Input, Table, Form, ButtonGroup, InputGroup, InputGroupAddon, Button, Row, Col} from 'reactstrap';
import CRUDSaveButton from '../CRUDButtonBar/CRUDSaveButton';
import CancelButton from '../buttons/CancelButton'
import * as CourseAPI from '../services/CourseAPI';
import Log from '../auxiliar/Log';

export default withRouter(function CalificationEdit(props){
    
    const [dirty, setDirty] = useState(false)
    const [califificationFullList, setCalifFullList] = useState(createCalifFullList(props.students, props.currEvalInstance.califications)) 
    const [filterValue, setFilterValue] = useState('');

    Log.info(califificationFullList, "Calification Picker")

    useEffect(() => {
        if (props.currEvalInstance){
            setCalifFullList(createCalifFullList(props.students, props.currEvalInstance.califications))
        }
    }, [props.currEvalInstance.califications])

    function handleNoteChange(event){
        setDirty(true);
        const newNote = event.target.value;
        setCalifFullList(califificationFullList.map( cal => 
            cal.student.fileNumber === parseInt(event.target.name) ? {...cal, note: newNote} :  cal))
    }

    function createCalifFullList(students, califications){
        // Uses student list and existing califications to create califications for all students
        if (props.currEvalInstance){
            const sl = students.map( st => {
                const calificationFound = califications.find( cal => cal.student.fileNumber === st.fileNumber);
                return ( populateCalif(st, calificationFound));
            })
            return sl;
        }    
        else {
            return [];
        }
    }

    function populateCalif (st, cal){
        if (cal){
            cal.student = st
            return cal;
        } else{
            return {
                calificationId: '',
                student: { ...st},
                note: ''
            }
        }
    }

    function onSaveCalifications(event){
        event.preventDefault()
        Log.info("Saving califications...........")
        const evalWithCalifications = {...props.currEvalInstance, califications: califificationFullList}
        CourseAPI.postCourseEvaluationAsync(
            props.courseId, 
            evalWithCalifications, 
            (res) => {
                Log.info(res, "POST SUCCEDED");
                setDirty(false);
                props.reloadCourse();

            }, 
            props.showError("save course")
        );

    }

    function isEvalSelected() { 
        return props.currEvalInstance && props.currEvalInstance.evaluationId !== '';
    }

    function getInnerPropValue(baseObj, subPropString){
        const subProps = subPropString.split(".");
        const lastPropName = subProps.pop(); // elimina del array y retorna el ultimo 
        let propRef = baseObj;
        subProps.forEach(subprop => {
          propRef = propRef[subprop];
        });
        return propRef[lastPropName];
    }

    const tableRender = (itemList) => {
        const filteredFields = ["student.personalData.lastName", "student.personalData.firstName", "student.fileNumber", "note"];
        let filteredList = itemList;
        if (props.currEvalInstance.evaluationId === ''){
            filteredList = []
        } else {

            if (filterValue !== '')
                filteredList = itemList.filter( cal => 
                    filteredFields.some( k => {
                        Log.info(getInnerPropValue(cal, k), "getInnerPropValue(cal, k)");
                        return getInnerPropValue(cal, k) ? getInnerPropValue(cal, k).toString().toLowerCase().includes(filterValue.toLowerCase()) : false
                    
                    })             
                )
        }
        return <Table hover className="mt-4" scrollable={"true"} data-testid="califTable" >
            <thead>
                <tr>
                    <th width="25%">File Number</th>
                    <th width="50%">Student Name</th>
                    <th width="25%">Grade</th>
                </tr>
            </thead>
            <tbody>
                {filteredList.map( (item, index) => 
                    <tr key={index} data-testid={`califTableRow_${index}`} >
                        <td style={{whiteSpace: 'nowrap'}}>{item.student.fileNumber || ''}</td>
                        <td style={{whiteSpace: 'nowrap'}}>{`${item.student.personalData.lastName}, ${item.student.personalData.firstName}` || ''}</td>
                        <td style={{whiteSpace: 'nowrap'}}>
                            <Input data-testid={`califTableRowInput_${index}`}
                                id={item.evaluationId}
                                name = {item.student.fileNumber}
                                value={item.note || ''} 
                                onChange={handleNoteChange}
                                disabled={!props.currEvalInstance}
                                type="float"  min="0.0" max="10.0"
                                title="A number between 0 and 10, with up to 2 decimals"
                                pattern={"^(10|10.0|10.00|([0-9])([.][0-9]([0-9])?)?)$"}
                                />
                        </td>
                    </tr>
                )}
            </tbody>
        </Table>;
    }
    
    
    return (
        <>
        <Form onSubmit={onSaveCalifications}>
            <Row>
                <Col md={10}>
                <InputGroup>
                    <Input 
                        placeholder={"Search student..."} 
                        value={filterValue} 
                        onChange={(event) => setFilterValue(event.target.value)} 
                        disabled = {!isEvalSelected()}
                        />
                    <InputGroupAddon addonType="append">
                        <Button color="secondary" onClick={() => setFilterValue('')} tabIndex={-1} >Clear</Button>
                    </InputGroupAddon>
                </InputGroup>
                </Col>
                <Col>
                    <ButtonGroup style={{ padding: "0em", float: "right"}}>
                    <CRUDSaveButton
                        behavior = {{type: "submit"}}
                        tabIndex={-1}
                        entityTypeCapName = {'Calification'}
                        isDisabled = {!isEvalSelected() || !dirty}
                        />
                    <CancelButton onClick={() => props.history.goBack()} />
                    </ButtonGroup>
                </Col>
            </Row>
            {tableRender(califificationFullList || [])}
        </Form>
        </>
    )
})
