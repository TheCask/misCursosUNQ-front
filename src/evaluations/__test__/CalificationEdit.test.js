import React from 'react';
import {render, fireEvent, cleanup } from '@testing-library/react';
import CalificationEdit from '../CalificationEdit';
import * as Constants from '../../auxiliar/Constants';


afterEach(cleanup);

it("renders correctly when no students are provided", () => {
    const {queryByTestId} = render(<CalificationEdit 
        courseId={1}//props.courseId}
        currEvalInstance = {{ evaluationId: 2}}// currEvalInstance}
        students = {[]}//currCourse.students}
        reloadCourse={null}
        showError={null}
        />
    )
    expect(queryByTestId("califTable")).toBeTruthy()
    expect(queryByTestId("califTableRow_0")).toBeFalsy()
})
    
it("renders correctly when there's a one-student list and no califications", () => {
    const {queryByTestId} = render(<CalificationEdit 
        courseId={1}//props.courseId}
        currEvalInstance = {{ evaluationId: 2, califications: []}}// currEvalInstance}
        students = {[Constants.emptyStudent]}//currCourse.students}
        reloadCourse={null}
        showError={null}
    />
    )
    expect(queryByTestId("califTable")).toBeTruthy()
    expect(queryByTestId("califTableRow_0")).toBeTruthy()
    expect(queryByTestId("califTableRow_1")).toBeFalsy()
    expect(queryByTestId("califTableRow_0").lastChild.nodeValue).toBeNull()
})
        

it("renders correctly when there's a two-student list and one calification", () => {
    const student1 = {...Constants.emptyStudent, fileNumber: 1}
    const student2 = {...Constants.emptyStudent, fileNumber: 2}
    const {queryByTestId, asFragment} = render(<CalificationEdit 
        courseId={1}//props.courseId}
        currEvalInstance = {{ evaluationId: 2, califications: [{calificationId: 1, student: student1, note: 10}] }}// currEvalInstance}
        students = {[student1, student2] }//currCourse.students}
        reloadCourse={null}
        showError={null}
    />
    )
    expect(asFragment()).toMatchSnapshot()
    expect(queryByTestId("califTable")).toBeTruthy()
    expect(queryByTestId("califTableRow_0")).toBeTruthy()
    expect(queryByTestId("califTableRow_1")).toBeTruthy()
    expect(queryByTestId("califTableRow_2")).toBeFalsy()
    expect(queryByTestId("califTableRowInput_0").value).toBe("10")
})