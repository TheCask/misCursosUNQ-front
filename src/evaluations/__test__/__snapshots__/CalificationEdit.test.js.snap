// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there's a two-student list and one calification 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="row"
    >
      <div
        class="col-md-10"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            placeholder="Search student..."
            type="text"
            value=""
          />
          <div
            class="input-group-append"
          >
            <button
              class="btn btn-secondary"
              tabindex="-1"
              type="button"
            >
              Clear
            </button>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="btn-group"
          role="group"
          style="padding: 0em; float: right;"
        >
          <button
            class="btn btn-primary btn-sm disabled"
            disabled=""
            id="save_Calification"
            tabindex="0"
            type="submit"
          />
        </div>
      </div>
    </div>
    <table
      class="mt-4 table table-hover"
      data-testid="califTable"
      scrollable="true"
    >
      <thead>
        <tr>
          <th
            width="25%"
          >
            File Number
          </th>
          <th
            width="50%"
          >
            Student Name
          </th>
          <th
            width="25%"
          >
            Grade
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          data-testid="califTableRow_0"
        >
          <td
            style="white-space: nowrap;"
          >
            1
          </td>
          <td
            style="white-space: nowrap;"
          >
            , 
          </td>
          <td
            style="white-space: nowrap;"
          >
            <input
              class="form-control"
              data-testid="califTableRowInput_0"
              max="10.0"
              min="0.0"
              name="1"
              pattern="^(10|10.0|10.00|([0-9])([.][0-9]([0-9])?)?)$"
              title="A number between 0 and 10, with up to 2 decimals"
              type="float"
              value="10"
            />
          </td>
        </tr>
        <tr
          data-testid="califTableRow_1"
        >
          <td
            style="white-space: nowrap;"
          >
            2
          </td>
          <td
            style="white-space: nowrap;"
          >
            , 
          </td>
          <td
            style="white-space: nowrap;"
          >
            <input
              class="form-control"
              data-testid="califTableRowInput_1"
              max="10.0"
              min="0.0"
              name="2"
              pattern="^(10|10.0|10.00|([0-9])([.][0-9]([0-9])?)?)$"
              title="A number between 0 and 10, with up to 2 decimals"
              type="float"
              value=""
            />
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</DocumentFragment>
`;
